### ISSUE TITLE
Coolprop Props function error

### ISSUE THREAD

**Issue by ragnar2015** [CoolProp/CoolProp#955]
I am currently using CoolProp version 3.2.0 with Mac OS X Mavericks 10.9.5

I am running into an error with this CoolProp call in Python:
tt = CoolProp.CoolProp.Props('T','H',h,'S',s,fluid)

fluid = 'co2'
The problematic h and s values are in the error trace below.

Any suggestions??
BTW I am kinda stuck with this version using the older style of function invocations.

Thanks in advance..

File "CoolProp.pyx", line 174, in CoolProp.CoolProp.Props (build/temp.macosx-10.9-intel-2.7/pyrex/CoolProp/CoolProp.cpp:11661)
File "CoolProp.pyx", line 288, in CoolProp.CoolProp.Props (build/temp.macosx-10.9-intel-2.7/pyrex/CoolProp/CoolProp.cpp:9967)
ValueError: Props failed ungracefully with inputs:"T",'H',-1.3644217765679036e+02,'S',-1.3117176167846665e+00,"co2"; please file a ticket at https://sourceforge.net/p/coolprop/tickets/

**Comment by ibell** [comment]
Sorry, but we don't support 3.2 any more - why are you using such an ancient version?  You should be able to update to at least 4.2.5. That's the basically the end of the version 4 (read, old) architecture, and version 5 is the beginning of the new architecture.

I tried the development version of CoolProp in IPython, and it didn't work either, which suggests that your inputs are invalid.  Where are your inputs? What temperature and pressure do they correspond to?
``` ipython
In [1]: import CoolProp

In [2]: CoolProp.CoolProp.PropsSI("T",'H',-1.3644217765679036e+05,'S',-1.3117176167846665e+03,"co2")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-8a1cabeebafc> in <module>()
----> 1 CoolProp.CoolProp.PropsSI("T",'H',-1.3644217765679036e+05,'S',-1.3117176167846665e+03,"co2")

CoolProp\CoolProp.pyx in CoolProp.CoolProp.PropsSI (CoolProp\CoolProp.cpp:32584)()

CoolProp\CoolProp.pyx in CoolProp.CoolProp.PropsSI (CoolProp\CoolProp.cpp:32366)()

CoolProp\CoolProp.pyx in CoolProp.CoolProp.__Props_err2 (CoolProp\CoolProp.cpp:30553)()

ValueError: Inputs in Brent [228.592000,1111.000000] do not bracket the root.  Function values are [12413148.445835,2380915.941598] : PropsSI("T","H",-136442.1777,"S",-1311.717617,"co2")
```

**Comment by jowr** [comment]
@ibell is right, you should update to version 4.2.5 (https://pypi.python.org/pypi/CoolProp/4.2.5) to do so, run 
```bash
pip uninstall CoolProp
pip install CoolProp==4.2.5
```

**Comment by ragnar2015** [comment]
Hi Ian,
In July 2014, I used homebrew to install Python 2.7.7 into my OSX 10.9.5 installation. Then I used pip to install CoolProp 4.2.2. It is located in directory: usr/local/lib/python2.7/site-packages/CoolProp. 

When I ran to the error that I posted about I was agast to discover that my code was actually running the 3.2 version from an installation in Dec 2013! I found CoolProp 3.2 installed in Library/Python/2.7/site-packages/CoolProp. 

I am not sure why my python is not accessing the correct CoolProp version (the 4.2 version). I looked at my first PATH entry: /usr/local/bin and found both unix commands: python and python2.7 that both link to python 2.7.7 in the homebrew "cellar". Any ideas?  I would like to move CoolProp 3.2 to an inactive location for now in case something breaks while establishing CoolProp 4.2.

BTW you mistyped the exponent of the entropy value into the your debug CoolProp call (should be e+00 not e+03). The -1.3117e+00 entropy for co2 should be reasonable. 

I definitely want to be running the 4.2 version.

**Comment by ragnar2015** [comment]
jowr, thanks for the syntax to pip install version 4.2.5. I might update after we can figure out why my python is not accessing my CoolProp 4.2 installation from July 2014.

**Comment by jowr** [comment]
You can simply run `pip uninstall CoolProp` many times until all versions of coolprop are gone. 

Your inputs seem to be wrong:
```Python
import CoolProp
print CoolProp.__version__
print CoolProp.CoolProp.PropsSI("T","H",-1.3644217765679036e+05,"S",-.3117176167846665e+00,"co2")
```
yields:
```
4.2.5
ValueError: CoolProp error: HS inputs are solid - not allowed h: -136442 s: -0.3
11718 :: inputs were:"T","H",-1.3644217765679036e+05,"S",-3.1171761678466647e-01
,"co2"
```

**Comment by jowr** [comment]
... and also this fails
```Python
import CoolProp
print CoolProp.__version__
print CoolProp.CoolProp.PropsSI("T","H",-1.3644217765679036e+05,"S",-1.3117176167846665e+00,"co2")
```

**Comment by ragnar2015** [comment]
My immediate concern is why my python is accessing coolprop 3.2 in "Library/Python/2.7/site-packages/CoolProp" instead of coolprop 4.2 in "usr/local/lib/python2.7/site-packages/CoolProp".
I used pip to install coolprop 4.2. There is some link or other that has not been established. Any ideas?

**Comment by ibell** [comment]
Hard to say.  Best to do as Jorrit said, keep running

pip uninstall CoolProp

until it says it can't uninstall CoolProp anymore.  Then, and only then, do

pip install CoolProp==4.2.5

Also, you should probably be using the Anaconda distribution of python (
https://www.continuum.io/downloads)

**Comment by ragnar2015** [comment]
Hi Ian,

Well, I deleted all CoolProp copies via pip uninstall CoolProp.

Then I attempted an install with "pip install CoolProp==4.2.5"

But I got a clang failed error with exit status 1    (12 warnings and 2 errors generated)

Here is the final error report:

12 warnings and 2 errors generated.

error: command 'clang' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/local/opt/python/bin/python2.7 -c "import setuptools, tokenize;__file__='/private/var/folders/80/1f0vyzts5jd87jjhmsr3pjmh0000gp/T/pip_build_carbon/CoolProp/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/80/1f0vyzts5jd87jjhmsr3pjmh0000gp/T/pip-SMQ2D2-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /private/var/folders/80/1f0vyzts5jd87jjhmsr3pjmh0000gp/T/pip_build_carbon/CoolProp

Help!!

**Comment by ragnar2015** [comment]
Hi Ian,

I googled around and found a response that you issued to a similar problem:
https://github.com/ibell/coolprop/issues/214

I removed an older version of cython from my system and CoolProp compiled with:
"Successfully installed CoolProp"
So far so good but...

I restarted the computer and opened a terminal session. 
Typed in "python"  (runs python 2.7.7)
Then I tried to "import CoolProp" but it failed because it could not find module "six": here is the trace:

Python 2.7.7 (default, Jun 14 2014, 23:12:13) 
[GCC 4.2.1 Compatible Apple LLVM 5.1 (clang-503.0.40)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import CoolProp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/CoolProp/__init__.py", line 7, in <module>
    from . import Plots
  File "/usr/local/lib/python2.7/site-packages/CoolProp/Plots/__init__.py", line 4, in <module>
    from .Plots import PropsPlot, IsoLines, drawIsoLines
  File "/usr/local/lib/python2.7/site-packages/CoolProp/Plots/Plots.py", line 5, in <module>
    import numpy, matplotlib, matplotlib.pyplot, math, re
  File "/usr/local/lib/python2.7/site-packages/matplotlib/pyplot.py", line 29, in <module>
    from matplotlib.figure import Figure, figaspect
  File "/usr/local/lib/python2.7/site-packages/matplotlib/figure.py", line 36, in <module>
    from matplotlib.axes import Axes, SubplotBase, subplot_class_factory
  File "/usr/local/lib/python2.7/site-packages/matplotlib/axes.py", line 20, in <module>
    import matplotlib.dates as _  # <-registers a date unit converter
  File "/usr/local/lib/python2.7/site-packages/matplotlib/dates.py", line 119, in <module>
    from dateutil.rrule import (rrule, MO, TU, WE, TH, FR, SA, SU, YEARLY,
  File "/usr/local/lib/python2.7/site-packages/dateutil/rrule.py", line 18, in <module>
    from six import advance_iterator, integer_types
ImportError: No module named six

I am updating python to a newer version of python 2.7 now. See what that does.

**Comment by ragnar2015** [comment]
I am now at python 2.7.11. The missing module "six" appears fixed.
Now the python import of CoolProp is failing again with "ImportError: No module named relativedelta" 

I checked the dateutil installation.. It looks OK. Here is the error trace.

pip install python-dateutil --upgrade
Requirement already up-to-date: python-dateutil in /usr/local/Cellar/matplotlib/1.5.0/libexec/lib/python2.7/site-packages
Requirement already up-to-date: six>=1.5 in /usr/local/Cellar/matplotlib/1.5.0/libexec/lib/python2.7/site-packages (from python-dateutil)


Python 2.7.11 (default, Jan 22 2016, 16:30:50) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import CoolProp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/CoolProp/__init__.py", line 7, in <module>
    from . import Plots
  File "/usr/local/lib/python2.7/site-packages/CoolProp/Plots/__init__.py", line 4, in <module>
    from .Plots import PropsPlot, IsoLines, drawIsoLines
  File "/usr/local/lib/python2.7/site-packages/CoolProp/Plots/Plots.py", line 5, in <module>
    import numpy, matplotlib, matplotlib.pyplot, math, re
  File "/usr/local/lib/python2.7/site-packages/matplotlib/pyplot.py", line 36, in <module>
    from matplotlib.figure import Figure, figaspect
  File "/usr/local/lib/python2.7/site-packages/matplotlib/figure.py", line 40, in <module>
    from matplotlib.axes import Axes, SubplotBase, subplot_class_factory
  File "/usr/local/lib/python2.7/site-packages/matplotlib/axes/__init__.py", line 4, in <module>
    from ._subplots import *
  File "/usr/local/lib/python2.7/site-packages/matplotlib/axes/_subplots.py", line 10, in <module>
    from matplotlib.axes._axes import Axes
  File "/usr/local/lib/python2.7/site-packages/matplotlib/axes/_axes.py", line 22, in <module>
    import matplotlib.dates as _  # <-registers a date unit converter
  File "/usr/local/lib/python2.7/site-packages/matplotlib/dates.py", line 129, in <module>
    from dateutil.relativedelta import relativedelta
ImportError: No module named relativedelta

Any ideas??

**Comment by ibell** [comment]
How did you get python?  I recommend (strongly!!) that you use Anaconda
distribution.  I've never had problems like these with anaconda.  You
should

conda install matplotlib

once you have Anaconda installed.

**Comment by ragnar2015** [comment]
Hi Ian,

I corresponded with you in April 2015 on another issue. Right now I am working GITHUB issue #955.

I had coolprop 3.2 on my Mac since Dec 2013 and wrote all of my code using kSI.

So enthalpy was KJ/kg and entropy was KJ/kg/K, pressure was kPa per the CoolProp 4.0.0 pdf documentation p54.

So today, per your advice, I removed coolprop 3.2 and finally was able to successfully install coolprop 4.2.5.

Python now imports it just fine.

But now all of my code has broken.

Does CoolProp 4.2.5 use kSI ??

I used to write per the CoolProp 4.0.0 pdf documentation:

p=35434.      # (kPa)		or 5140 psia
t = 363.3		# (deg K)	or 194.3 deg F
h = CoolProp.CoolProp.Props('H','P',p,'T',t,'co2')

and get h = -63.951 kJ/kg (note the minus)   using CoolProp 3.2
entropy = -0.3315 kJ/kg/K


Now with Coolprop 4.2.5, I get h=364.445 but I do not know what the units are now.

Was there a units default change somewhere???

And thank you for your help.

Don Parker

**Comment by ragnar2015** [comment]
Hi Ian,

My latest code that I was working on is broken for some reason with my new CoolProp 4.2.4 (was 3.2)
That code is dancing around the saturation dome in co2 so there is some problem  there. 

However, a very important code that I worked on earlier still works! So that is good.
I have not checked everything but that is my status so far.

I have to figure out what is happening with my current projects.

With regards to that dateutil problem, I had to force a dateutil version 1.5  to my computer.

That worked!

Don

**Comment by jowr** [comment]
@ragnar2015, I am happy that it almost works for you. However, I encourage you to take the step to convert your code to version 5. There are only very few users of version 4 left and you might not get that much support...

**Comment by ragnar2015** [comment]
Hi Jorritt,

Thank you on the heads up on v4 vs v5.

The reason I have resisted moving to verson 5 was that the PropSI function uses pure SI units (Pascal, Joules etc) whereas I wrote lots of code with the default kSI calls (kPa, kJ/jg etc). Is there a flag in version 5 to use kSI?  

Don

**Comment by ragnar2015** [comment]
...just for the record..

1) I updated python 2.7.7 to 2.7.11. This cleared the exception "ImportError: No module named six" raised when I tried to import CoolProp.

2) Then I tried to import CoolProp again and got another exception: 
ImportError: No module named relativedelta
I fixed this by forcing a pip install of dateutil version 1.5

So now I have CoolProp 4.2.5 working. 

But now, there are several of my programs that crash with 4.2.5. They ran OK with CoolProp 3.2.
I will launch a separate ticket on at least one problem I see already. Thanks for your help thus far!
